src/replication.c

```diff
@@ -3520,13 +3602,29 @@ void processClientsWaitingReplicas(void) {
             unblockClient(c);
             addReplyLongLong(c,last_numreplicas);
         } else {
-            int numreplicas = replicationCountAcksByOffset(c->bstate.reploffset);
+            int numreplicas = c->bstate.btype == BLOCKED_WAITAOF ?
+                replicationCountAOFAcksByOffset(c->bstate.reploffset) :
+                replicationCountAcksByOffset(c->bstate.reploffset);
 
             if (numreplicas >= c->bstate.numreplicas) {
                 last_offset = c->bstate.reploffset;
                 last_numreplicas = numreplicas;
+
+                /* Check if the local constraint of WAITAOF is served */
+                int wait_aof = c->bstate.btype == BLOCKED_WAITAOF;
+                int localok = server.fsynced_reploff >= c->bstate.reploffset;
```

Comment:
Maybe name this `numlocal` for consistency?
```suggestion
                int numlocal = server.fsynced_reploff >= c->bstate.reploffset;
```